README.md

```diff
@@ -179,13 +179,101 @@ All the widgets will have the same size, which is chosen based on the number of
 Custom metric monitoring can be created for simple metrics, simple metrics with anomaly detection and search metrics.
 The first two also support alarming.
 
-### Custom monitoring
+#### Example: simple metrics
+
+```ts
+monitorCustom({
+    title: "Three simple metrics",
+    metrics: [m1, m2, m3]
+})
+```
+
+Adding an alarm:
+
+```ts
+monitorCustom({
+    title: "One simple metric with alarm",
+    metrics: [
+        {
+            metric: m1,
+            alarmFriendlyName: "MetricWithAlarm",
+            addAlarm: {
+                Warning: {
+                    threshold: 1,
+                    comparisonOperator: ComparisonOperator.GREATER_THAN_THRESHOLD
+                }
+            }
+        }
+    ]
+})
+```
+
+#### Example: metric with anomaly detection
+
+In this case, only one metric is supported. 
+There cannot be multiple metrics rendered with anomaly detection in a single widget (CloudWatch limitation).
+
+```ts
+monitorCustom({
+    title: "Metric with anomaly detection",
+    metrics: [
+        {
+            metric: m1,
+            anomalyDetectionStandardDeviationToRender: 3
+        }
+    ]
+})
+```
+
+Adding an alarm:
+
+```ts
+monitorCustom({
+    title: "Metric with anomaly detection and alarm",
+    metrics: [
+        {
+            metric: m1,
+            alarmFriendlyName: "MetricWithAnomalyDetectionAlarm",
+            anomalyDetectionStandardDeviationToRender: 3,
+            addAlarmOnAnomaly: {
+                Warning: {
+                    standardDeviationForAlarm: 4,
+                    alarmWhenAboveTheBand: true,
+                    alarmWhenBelowTheBand: true
+                }
+            }
+        }
+    ]
+})
+```
+
+#### Example: search metrics
+
+```ts
+monitorCustom({
+    title: "Metric search",
+    metrics: [
+        {
+            searchQuery: "My.Prefix.",
+            dimensions: {
+                FirstDimension: "FirstDimensionValue",
+                // allow any value for the given dimension
+                SecondDimension: undefined
```

Comment:
This would have to be `undefined as unknown as string` to satisfy the type of `DimensionsMap`.

Reply:
Wow. That seems bad. We should probably introduce some magical constant instead, like `AnyDimensionValue`.